assets/js/googlesitekit/data/create-notifications-store.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * Provides API functions to create a datastore for notifications.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import { keyBy, omit } from 'lodash';
+import { get } from 'googlesitekit-api';
+
+const ADD_NOTIFICATION = 'ADD_NOTIFICATION';
+const REMOVE_NOTIFICATION = 'REMOVE_NOTIFICATION';
+const FETCH_NOTIFICATIONS = 'FETCH_NOTIFICATIONS';
+const RECEIVE_NOTIFICATIONS = 'RECEIVE_NOTIFICATIONS';
+const RECEIVE_NOTIFICATIONS_FAILED = 'RECEIVE_NOTIFICATIONS_FAILED';
+
+// This should remain private for now, hence not be exported on Data.
+export const createNotificationsStore = ( type, identifier, datapoint ) => {
+	const INITIAL_STATE = {
+		serverNotifications: undefined,
+		clientNotifications: {},
+		isFetchingNotifications: false,
+	};
+
+	const actions = {
+		addNotification( notification ) {
+			return {
+				notification,
```

Comment:
Okay, so this is different to what Gutenberg uses then. It makes sense to me though. We should stick to that then for any actions though I'd say? And we should add it to the guidelines doc.

Reply:
Yeah, I'm not a fan of the Gutenberg approach. It's less extensible, if we want to change payloads in the future it requires more refactoring, and because some payloads will require extra flags but others won't, they'll be inconsistent.

I think we should stick to the approach of not having payload directly like this, yes. I added a note about this to the guidelines 👍